src/test/system_tests/test_cloud_pools.js

```diff
@@ -212,152 +211,150 @@ async function verify_object_parts_on_cloud_nodes(replicas_in_tier, bucket_name,
 function run_test() {
     let replicas_in_tier;
     let files_bucket_tier;
-    return authenticate()
-        .then(function() {
-            return init_s3();
-        })
-        .then(() => client.account.add_external_connection({
-            name: TEST_CTX.connection_name,
-            endpoint: 'https://s3.amazonaws.com',
-            identity: process.env.AWS_ACCESS_KEY_ID,
-            secret: process.env.AWS_SECRET_ACCESS_KEY,
-            endpoint_type: 'AWS'
-        }))
-        .then(() => client.pool.create_cloud_pool({
-            name: TEST_CTX.cloud_pool_name,
-            connection: TEST_CTX.connection_name,
-            target_bucket: TEST_CTX.target_bucket
-        }))
-        .then(() => client.bucket.read_bucket({
-            name: TEST_CTX.source_bucket
-        }))
-        .then(function(source_bucket) {
-            const tier_name = source_bucket.tiering.tiers[0].tier;
-            return client.tier.read_tier({
-                    name: tier_name
-                })
-                .then(function(tier) {
-                    replicas_in_tier = tier.chunk_coder_config.replicas;
-                    files_bucket_tier = tier;
-                    const new_pools = tier.attached_pools.concat(TEST_CTX.cloud_pool_name);
-                    return client.tier.update_tier({
-                        name: tier.name,
-                        attached_pools: new_pools,
-                        data_placement: 'MIRROR'
-                    });
-                })
-                .then(function() {
-                    return client.tier.read_tier({
-                            name: tier_name
-                        })
-                        .then(function(tier) {
-                            replicas_in_tier = tier.chunk_coder_config.replicas;
-                            files_bucket_tier = tier;
-                        });
+    authenticate();
```

Comment:
I suggest replacing this with async/await on the way. It will clear the confusion with the promises and what the functions return